pandas/core/computation/eval.py

```diff
@@ -333,6 +335,16 @@ def eval(
 
         parsed_expr = Expr(expr, engine=engine, parser=parser, env=env)
 
+        if (
+            is_extension_array_dtype(parsed_expr.terms.return_type)
+            or getattr(parsed_expr.terms, "operand_types", None) is not None
+            and any(
+                is_extension_array_dtype(elem)
+                for elem in parsed_expr.terms.operand_types
+            )
+        ):
+            engine = "python"
```

Comment:
Does this mean if a user is specifying `engine="numexpr"`, this switches the engine to `python`?

If so, I think it would be better to explicitly raise a `NotImplementedError` and tell the user to switch the engine manually.

Reply:
Yep, would you be ok with a RuntimeWarning? Since numexpr is the default, raising seems a bit noisy maybe?

Reply:
Well the issues linked already raise an exception currently, so a `NotImplementedError` would be similar but more explicit. Unless there's a case I'm not thinking of.

Reply:
I am more concerned with people using our nullable option who are switching over from NumPy. Would be annoying if query/eval stop working then.

But you are correct, an appropriate NotImplementedError would be an improvement

Reply:
That's fair. Yeah I `RuntimeWarning` warning & encouraging users to switch to the python engine would be okay

Reply:
Added the warning